flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```
@@ -530,54 +515,109 @@ Actor with ActorLogMessages with ActorLogging {
-    val jobManagerSystem = AkkaUtils.createActorSystem(configuration, listeningAddress)
 
-    startActor(Props(new JobManager(configuration) with WithWebServer))(jobManagerSystem)
+  def runJobManager(configuration: Configuration,
+                    executionMode: ExecutionMode,
+                    listeningAddress: Option[(String, Int)]) : Unit = {
+
+    LOG.info("Starting JobManager")
+    LOG.debug("Starting JobManager actor system")
 
-    if(executionMode.equals(LOCAL)){
-      TaskManager.startActorWithConfiguration("", configuration,
-        localAkkaCommunication = false, localTaskManagerCommunication = true)(jobManagerSystem)
+    val jobManagerSystem = try {
+      AkkaUtils.createActorSystem(configuration, listeningAddress)
     }
+    catch {
+      case t: Throwable => {
+        if (t.isInstanceOf[org.jboss.netty.channel.ChannelException]) {
+          val cause = t.getCause()
+          if (cause != null && t.getCause().isInstanceOf[java.net.BindException]) {
+            val address = listeningAddress match {
+              case Some((host, port)) => host + ":" + port
+              case None => "unknown"
+            }
 
-    jobManagerSystem.awaitTermination()
+            throw new Exception("Unable to create JobManager at address " + address + ": " + cause.getMessage(), t)
+          }
+        }
+        throw new Exception("Could not create JobManager actor system", t)
+      }
+    }
+
+    try {
+      LOG.debug("Starting JobManager actor")
+
+      startActor(configuration, jobManagerSystem)
+
+      if(executionMode.equals(LOCAL)){
+        LOG.info("Starting embedded TaskManager for JobManager's LOCAL mode execution")
+
+        TaskManager.startActorWithConfiguration("", configuration,
+          localAkkaCommunication = false, localTaskManagerCommunication = true)(jobManagerSystem)
+      }
+
+      jobManagerSystem.awaitTermination()
+    }
+    catch {
+      case t: Throwable => {
+        Try(jobManagerSystem.shutdown())
```

Do you want to swallow all potential exceptions of ```ActorSystem.shutdown``` intentionally?